dask/dataframe/io/parquet.py

```diff
@@ -143,54 +143,62 @@ def _meta_from_dtypes(to_read_columns, file_dtypes, index_cols,
     return df
 
 
-def _normalize_columns(user_columns, data_columns):
-    """Normalize user- and file-provided column names
+def _normalize_index_columns(user_columns, data_columns, user_index, data_index):
+    """Normalize user and file-provided column and index names
 
     Parameters
     ----------
     user_columns : None, str or list of str
     data_columns : list of str
+    user_index : None, str, or list of str
+    data_index : list of str
 
     Returns
     -------
     column_names : list of str
+    index_names : list of str
     out_type : {pd.Series, pd.DataFrame}
     """
-    out_type = DataFrame
-    if user_columns is not None:
-        if isinstance(user_columns, string_types):
-            columns = [user_columns]
-            out_type = Series
-        else:
-            columns = list(user_columns)
-        column_names = columns
-    else:
-        column_names = list(data_columns)
-    return column_names, out_type
+    specified_columns = user_columns is not None
+    specified_index = user_index is not None
 
+    out_type = DataFrame
 
-def _normalize_index(user_index, data_index):
-    """Normalize user- and file-provided index names
-
-    Parameters
-    ----------
-    user_index : None, str, or list of str
-    data_index : list of str
-
-    Returns
-    -------
-    index_names : list of str
-    """
-    if user_index is not None:
-        if user_index is False:
-            index_names = []
-        elif isinstance(user_index, string_types):
-            index_names = [user_index]
-        else:
-            index_names = list(user_index)
+    if user_columns is None:
```

Comment:
It might be useful to have some comments to describe the rationale behind what this is doing at each step and why.

Reply:
Done.